### PR TITLE
Base+LibWeb: Add basic implementation of progress bar element

### PR DESCRIPTION
**Pull request by Sauler**
Many things still has to be done, but i think it's good starting point.

### CODE REVIEW

**Review by linusg — CHANGES_REQUESTED** [review]
https://github.com/SerenityOS/serenity/blob/master/CONTRIBUTING.md

> Write the commit message subject line in the imperative mood ("Foo: Change the way dates work", not "Foo: Changed the way dates work").

**Comment by trflynn89** [comment]
Some general questions/comments:
1. Do we actually need the float parser? Does it give different results than using `strtod`?
2. When adding new elements like this, the PR should also include a test page under Base/res/html/misc.
3. Commits should all be atomic - in the PR's current state, there's some messy things like adding a `dbgln` call in one commit, then removing it in the next commit. That removal should be squashed against the commit that added it such that the `dbgln` was never present in the first place.

File: Userland/Libraries/LibWeb/HTML/HTMLProgressElement.cpp [hunk 1]
```diff
@@ -17,4 +19,57 @@ HTMLProgressElement::~HTMLProgressElement()
...
+
+    set_attribute(HTML::AttributeNames::value, String::formatted("{}", value));
+
+    if (layout_node())
+        layout_node()->set_needs_display();
+}
+
+double HTMLProgressElement::max() const
+{
+    auto parsedValue = strtod(attribute(HTML::AttributeNames::max).characters(), nullptr);
```
Comment (drunderscore) [inline Userland/Libraries/LibWeb/HTML/HTMLProgressElement.cpp:49]: All variables should use `snake_case`

File: Userland/Libraries/LibWeb/HTML/HTMLProgressElement.cpp [hunk 2]
```diff
@@ -17,4 +19,57 @@ HTMLProgressElement::~HTMLProgressElement()
...
 }
 
+RefPtr<Layout::Node> HTMLProgressElement::create_layout_node(NonnullRefPtr<CSS::StyleProperties> style)
+{
+    return adopt_ref(*new Layout::Progress(document(), *this, move(style)));
+}
+
+double HTMLProgressElement::value() const
+{
+    auto parsedValue = strtod(attribute(HTML::AttributeNames::value).characters(), nullptr);
```
Comment (drunderscore) [inline Userland/Libraries/LibWeb/HTML/HTMLProgressElement.cpp:29]: All variables should use `snake_case`

File: Userland/Libraries/LibWeb/HTML/HTMLProgressElement.cpp [hunk 3]
```diff
@@ -17,4 +19,57 @@ HTMLProgressElement::~HTMLProgressElement()
...
+
+    return parsedValue;
+}
+
+void HTMLProgressElement::set_max(double value)
+{
+    if (value <= 0)
+        return;
+
+    set_attribute(HTML::AttributeNames::max, String::formatted("{}", value));
```
Comment (drunderscore) [inline Userland/Libraries/LibWeb/HTML/HTMLProgressElement.cpp:61]: This could use `String::number` (even though it does the exact same thing as this does)

**Review by drunderscore — COMMENTED** [review]
Just a few stylistic problems! :^)

File: Userland/Libraries/LibWeb/HTML/HTMLProgressElement.cpp (hunk 1 above)
Comment (Sauler) [reply Userland/Libraries/LibWeb/HTML/HTMLProgressElement.cpp:49]: Forgot about that :D

**Review by trflynn89 — APPROVED** [review]
Nice, this looks much better :D

File: Userland/Libraries/LibWeb/HTML/HTMLProgressElement.cpp [hunk 4]
```diff
@@ -17,4 +19,57 @@ HTMLProgressElement::~HTMLProgressElement()
...
+
+    return min(parsed_value, max());
+}
+
+void HTMLProgressElement::set_value(double value)
+{
+    if (value < 0)
+        return;
+
+    set_attribute(HTML::AttributeNames::value, String::formatted("{}", value));
```
Comment (drunderscore) [inline Userland/Libraries/LibWeb/HTML/HTMLProgressElement.cpp:41]: Sorry to nitpick, but I did miss commenting on this stringify, my bad :^(

Comment (Sauler) [reply Userland/Libraries/LibWeb/HTML/HTMLProgressElement.cpp]: I should check all places for this. Fixed :)

**Pull request merged**